### ISSUE TITLE
Crash on data asset.(4.26.2, BA 3.0.2)

### ISSUE THREAD

**Issue by yuanb36** [fpwong/BlueprintAssistWiki#41]
Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x0000000000000018

UE4Editor_BlueprintAssist!FBAAssetEditorHandler::OnAssetClosed() [C:\Git Repositories\sas\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistAssetEditorHandler.cpp:140]
UE4Editor_BlueprintAssist!FBAAssetEditorHandler::CheckInvalidAssetEditors() [C:\Git Repositories\sas\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistAssetEditorHandler.cpp:176]
UE4Editor_BlueprintAssist!FBAInputProcessor::Tick() [C:\Git Repositories\sas\Plugins\BlueprintAssist\Source\BlueprintAssist\Private\BlueprintAssistInputProcessor.cpp:353]
UE4Editor_Slate!FSlateApplication::FinishedInputThisFrame() [D:\Build\++UE4\Sync\Engine\Source\Runtime\Slate\Private\Framework\Application\SlateApplication.cpp:1317]
UE4Editor!FEngineLoop::Tick() [D:\Build\++UE4\Sync\Engine\Source\Runtime\Launch\Private\LaunchEngineLoop.cpp:4814]
UE4Editor!GuardedMain() [D:\Build\++UE4\Sync\Engine\Source\Runtime\Launch\Private\Launch.cpp:169]
UE4Editor!GuardedMainWrapper() [D:\Build\++UE4\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:137]
UE4Editor!WinMain() [D:\Build\++UE4\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:268]
UE4Editor!__scrt_common_main_seh() [d:\agent\_work\5\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288]
kernel32
ntdll

To produce: 
1. Create a fresh new project.
2. Create a blueprint derived from PrimaryDataAsset.
3. Create a data asset of previously defined  PrimaryDataAsset type.
4. In the data asset BP, define some variables.
5. Define a function, catenate some Print String nodes, auto arrange them.
6. Open the data asset, hit compile in the data asset BP, if not crash, close the BP, open it and compile again. Or try to close and open the data asset some times.
7. The crash will randomly happens with data asset opened while you compiling the data asset BP or try to open or close it.

**Comment by fpwong** [comment]
Thanks yuanb36, can confirm this causes a crash. 

It will always crash if you have your your PrimaryDataAsset open in editor while compiling your base DataAsset BP. Will look into this.

**Comment by fpwong** [comment]
Hi yuanb36, while I did stop the crash occuring this issue seems to be due to a bug in the engine code. So I have to check if the UObject is valid using IsValidLowLevelFast and when this fails it will log it in the output log (since it's never meant to fail). As a result if you trigger this bug to occur using the steps above, it will print *a lot* of messages.

To be safe I suggest you close your DataAsset from step 3. before compiling your blueprint data asset as even running certain editor functionality may crash doing stuff not related to the plugin. This will also stop the debug messages from occuring.

I've submitted a pull request to fix the cause of issue: https://github.com/EpicGames/UnrealEngine/pull/8164
